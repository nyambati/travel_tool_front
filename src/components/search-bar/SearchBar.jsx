import React, { PureComponent } from 'react';

import Loupe from '../../images/Loupe.svg';
import './SearchBar.scss';


/**
 * @description - Contains search bar
 *
 * @class SearchBar
 *
 * @extends {PureComponent}
 *
 */


class SearchBar extends PureComponent {
  render() {
    return (
      <div className="mdl-search">
        <img src={Loupe} alt="Search Icon" className="mdl-search-icon" />
        <form action="#">
          <input
            className="input-format"
            type="text"
            placeholder="Search requests, team members, etc"
            size="1"
          />
        </form>
      </div>
    );
  }
}
export default SearchBar;
