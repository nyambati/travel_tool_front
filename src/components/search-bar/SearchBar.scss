.mdl-search {
  margin: 0 100px;
  align-items: center;
  justify-content: center;
}

.mdl-search form {
  height: 36px;	
  width: 360px;	
  border-radius: 18px;	
  background-color: #F8F8F8;
  margin-top: 15px;	

}

.mdl-search,
.mdl-search form,
.mdl-search input {
  /* Cross browser flex-grow */
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.mdl-search,
.mdl-search form {
  /* Cross browser inline-flex */
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

/* Position search icon */
.mdl-search-icon {
  position: relative;
  left: 340px;
  height: 15.99px;	
  width: 16.05px;	
  margin-top: 15px;
}

.mdl-search input {
  border: none;
  border-radius: 18px;
  font-size: 16px;
  color: #999999;	
  background-color: #F8F8F8;
  padding: 0px 5px 0px 20px;
  line-height: 18px; 
  font-family: "DIN Pro";
  font-size: 14px;
  height: 36px;	
  width: 360px;
}



