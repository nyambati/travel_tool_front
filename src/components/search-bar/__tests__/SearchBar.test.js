import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../SearchBar';

// describe what we are testing
describe("Render SearchBar component", () => {
  // make our assertions and what we expect to happen
  it("should match snapshot", () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the searchbar as expected", () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper.length).toBe(1);
  });
});
