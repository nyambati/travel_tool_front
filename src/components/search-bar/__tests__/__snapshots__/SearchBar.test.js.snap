// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SearchBar component should match snapshot 1`] = `
<div
  className="mdl-search"
>
  <img
    alt="Search Icon"
    className="mdl-search-icon"
    src="Loupe.svg"
  />
  <form
    action="#"
  >
    <input
      className="input-format"
      placeholder="Search requests, team members, etc"
      size="1"
      type="text"
    />
  </form>
</div>
`;
